src/jsonrpc/api.rs

```diff
@@ -87,6 +87,41 @@ fn broadcast_spend(control: &DaemonControl, params: Params) -> Result<serde_json
     Ok(serde_json::json!({}))
 }
 
+fn list_coins(control: &DaemonControl, params: Option<Params>) -> Result<serde_json::Value, Error> {
+    let statuses_arg = params
+        .clone()
+        .and_then(|p| p.get(0, "statuses").cloned())
+        .and_then(|statuses| statuses.as_array().cloned());
```

Comment:
That's a lot of `clone()`s. You don't need to allocate here since you just go through it.

```diff
diff --git a/src/jsonrpc/api.rs b/src/jsonrpc/api.rs
index 1432428..f2b2f4b 100644
--- a/src/jsonrpc/api.rs
+++ b/src/jsonrpc/api.rs
@@ -89,9 +89,9 @@ fn broadcast_spend(control: &DaemonControl, params: Params) -> Result<serde_json
 
 fn list_coins(control: &DaemonControl, params: Option<Params>) -> Result<serde_json::Value, Error> {
     let statuses_arg = params
-        .clone()
-        .and_then(|p| p.get(0, "statuses").cloned())
-        .and_then(|statuses| statuses.as_array().cloned());
+        .as_ref()
+        .and_then(|p| p.get(0, "statuses"))
+        .and_then(|statuses| statuses.as_array());
     let mut statuses: Vec<CoinStatus> = vec![];
     if let Some(statuses_arg) = statuses_arg {
         for status_arg in statuses_arg.iter() {
@@ -104,8 +104,9 @@ fn list_coins(control: &DaemonControl, params: Option<Params>) -> Result<serde_j
         }
     }
     let outpoints = params
-        .and_then(|p| p.get(1, "outpoints").cloned())
-        .and_then(|op| op.as_array().cloned())
+        .as_ref()
+        .and_then(|p| p.get(1, "outpoints"))
+        .and_then(|op| op.as_array())
         .and_then(|arr| {
             arr.iter()
                 .map(|entry| {
```

Reply:
Thanks for this :pray: Have updated it now.